kernel/src/lib.rs

```diff
@@ -75,6 +75,7 @@ pub mod snapshot;
 pub mod table;
 pub(crate) mod utils;
 
+pub use delta_kernel_derive;
```

Comment:
@scovich  I wanted to flag this. I had to do this to do this because the `column_name` macro (used by `column_expr`) uses the macro from `delta_kernel_derive`.

Reply:
it's a separate crate so I think we can tackle this two ways?
1. (this approach) we can export it via kernel since it's a dependency of kernel or
2. we can just import it as a separate crate into FFI crate right?

seems like better separation of concerns to just to (2) but since kernel will always depend on delta_kernel_derive, I could also be okay with option (1).

curious to hear from @nicklan

Reply:
> It seems ok? But I'll admit I don't know the implications of making that crate public?

AFAIK the crate is already [public](https://crates.io/crates/delta_kernel_derive), and this is just re-exporting it as a public sub-crate from kernel.

Reply:
It feels like it also makes sense to do 1 from a usability standpoint. Suppose delta-rs wants to create columns using `column_expr!` or `column_name!`. Without the re-export, they'd get a compilation error coming from the macro, and they'd have to go through a couple layers of macro before finding *why* it fails.